src/main/java/com/hankcs/algorithm/AhoCorasickDoubleArrayTrie.java

```diff
@@ -114,6 +114,33 @@ public void parseText(String text, IHit<V> processor)
      * @param text The text
      * @param processor A processor which handles the output
      */
+    public void parseText(String text, IHitCancellable<V> processor)
+    {
+        int position = 1;
+        int currentState = 0;
+        for (int i = 0; i < text.length(); ++i)
```

Comment:
Was the `++i` instead of `i++` intentional? They have different behaviours.

Reply:
I don't know what the original author intended (the new function is an almost exact copy of the ones already there), but in this context, `++i` and `i++` are equivalent, because the result of that expression is not used.

Reply:
That's true. I would change this to `i++` though, that's the norm and leaves no space for confusing for future readers.